src/Factory/AwsFactory.php

```diff
@@ -16,18 +15,20 @@ class AwsFactory implements FactoryInterface
 {
     /**
      * @param ContainerInterface $container
-     * @param string             $requestedName
-     * @param array|null         $options
+     * @param string $requestedName
+     * @param array|null $options
      * @return AwsSdk
      */
     public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         // Instantiate the AWS SDK for PHP
         $config = $container->get('Config');
         $config = isset($config['aws']) ? $config['aws'] : [];
+
+        $zend_framework_version = isset($config['aws_zf2']['zend_framework_version']) ? $config['aws_zf2']['zend_framework_version'] : '3.0';
         $config += [
             'ua_append' => [
-                'ZF2/' . Version::VERSION,
+                'ZF2/' . $zend_framework_version,
```

Comment:
Thank you for sharing your thoughts! I think we are on the same side to make this backwards compatible and do not break other customers. I definitely see your efforts in making the change lives on `v2.1`. I'd prefer have a major version bump instead since this is supporting `zf3` instead of a bug fix, then I'll do the `README.md` work of course :). I'll take a look at your latest commit.